lib/sender.js

```diff
@@ -27,6 +27,24 @@ class Sender {
     this._bufferedBytes = 0;
     this._deflating = false;
     this._queue = [];
+
+    this._socket.prependOnceListener('close', () => {
```

Comment:
Do we need `prependOnceListener()`? Or is it used only to ensure that the listener is added before the one added in `websocket.js`? In the latter case I would just add a comment in `WebSocket#setSocket()` to specify that the `Sender` must be instantiated before adding the `'close'` listener to the `net.Socket`.

Reply:
Turns out we do actually need it.  I was testing a different ready state in my initial pass.

Reply:
Hmm ok, that's strange as this is added before the other anyway. I'm not a fan of `prependOnceListener()` because it will add the listener before all other listeners including the ones added by Node.js itself.

Reply:
Let me poke it again, I may end up blaming jet lag. :D

Reply:
Yep, going to blame jet lag.  Sorry for the confusion.  We do not need prepend, I'll switch it over to the comment instead.  The issue was I was switching `prepend` to `once` in the Sender but not in the MockSocket for the test.